vizro-ai/pyproject.toml

```diff
@@ -16,7 +16,7 @@ dependencies = [
   "pandas",
   "tabulate",
   "openai>=0.27.8,<1.0.0",  # TODO add support for openai>=1.0.0
-  "langchain==0.0.325",
+  "langchain==0.0.329",
```

Comment:
Can't we just do:

```suggestion
  "langchain>=0.0.329",
```

This has been raised by Snyk several times now always requiring the latest version. Is there a reason why we would want to pin to a specific version?

Reply:
I think there are 2 reasons on why we ended up with pin a specific langchain version.
1. Due the langchain is a extremely active (release many times in a week) and thus less stable library, we would always expect to see Snyk found an vulnerability from an older version and ask us to bump to the latest. So even if we do `langchain>=0.0.329`, few weeks later we might need to do something like `langchain>=0.0.345`
2. Also due to langchain is not stable, we want to be sure the version we allow won't break current Vizro-AI. Other than unit tests and integration tests we have, there is another step of user testing (manually run the example and check whether the final output make sense)

Reply:
> 1. In this case I would actually expect that Snyk would not complain anymore, because `langchain>=0.0.329` would automatically install the latest version and thus have the newest security upgrades included?

Snyk still complains in this case. Say if the latest version is already 0.0.400 and it found an issue in version 0.0.330, which might be fixed in 0.0.399. It will suggest a version bump to langchain>=0.0.399 still. Basically to enforce users to avoid installing versions that are vulnerable.

 

> 2\. I understand that langchain is not stable and then we don't want to run into the risk of breaking the current vizro-ai version, but at the same time we won't be able to identify non-working future versions if we pin to a specific older version. I think we should do something like this `langchain>=0.0.329, <0.1.0` in case we consciously want to upgrade, but pinning to a specific version seems too limiting. WDYT?

I'm quite open to this argument actually. As long as we expect there might be unexpected changes, but the benefit is providing a wider range of versions. @maxschulz-COL , are you aware of any other reasons why we pin to a specific version?

Reply:
I think @Anna-Xiong had quite some extensive difficulties with langchain changing in fundamental ways even between these patch versions. So this is the main reason I can remember why we are pinning the version. At times it was as bad as things breaking several times a week.

Given that we currently have somewhat lower scope to maintain/keep up with breaking changes in langchain I would be cautious in opening up the upper bound. If we ever get the request, it is easy to simply bump to the latest version, run automatic and manual tests, and then bump the dependency (or not). What we gain is a more stable package that does not require constant maintenance and further releases.